src/opnsense/mvc/app/views/OPNsense/Diagnostics/log.volt

```diff
@@ -99,18 +129,64 @@
               params.push("searchPhrase=" + encodeURIComponent($("input.search-field").val()));
           }
           if ( $('#severity_filter').val().length > 0) {
-              params.push("severity=" + encodeURIComponent($('#severity_filter').val().join(",")));
+              let r_severity = filter_exact ? $('#severity_filter').val() : severities.slice(0,severities.indexOf($('#severity_filter').val()) + 1);
+              params.push("severity=" + encodeURIComponent(r_severity.join(",")));
           }
           if (params.length > 0) {
               download_link = download_link + "?" + params.join("&");
           }
           $('<a></a>').attr('href',download_link).get(0).click();
       });
+
       updateServiceControlUI('{{service}}');
 
       // move filter into action header
       $("#severity_filter_container").detach().prependTo('#grid-log-header > .row > .actionBar > .actions');
+
+
+      function switch_mode(value) {
+          let select = $("#severity_filter");
+
+          // switch select mode and destroy selectpicker
+          select.prop("multiple", filter_exact);
+          select.selectpicker('destroy');
+
+          // remove title option. bug in bs-select. fixed in v1.13.18 https://github.com/snapappointments/bootstrap-select/issues/2491
+          select.find('option.bs-title-option').remove();
+
+          let header_val = filter_exact ? m_header : s_header;
+          select.selectpicker({ header: header_val });
+
+          // attach event handler each time header created
+          $("#exact_severity").on("click", function(event) {
+              event.stopPropagation();
+              filter_exact = !filter_exact;
+              let select = $("#severity_filter");
+
+              // set new select value to current value or highest value of multiselect
+              let new_val = Array.isArray(select.val()) ? select.val().pop() : select.val();
+
+              if (window.localStorage) {
+                  if (filter_exact) {
+                      localStorage.setItem('log_filter_exact_{{module}}_{{scope}}', 1);
+                  } else {
+                      localStorage.removeItem('log_filter_exact_{{module}}_{{scope}}');
+                  }
+                  // store user choice
+                  localStorage.setItem('log_severity_{{module}}_{{scope}}', new_val);
+              }
+              switch_mode(new_val);
+              // keep it open
+              select.selectpicker('toggle');
```

Comment:
For some reason _toggle_ does not work. The menu is still opened after each page reload. I've tested it again with _open_ and _refresh_, there it works.😮
I'm not too deep into bootstrap-select to tell why... 

```suggestion
              select.addClass('open');
              select.selectpicker('refresh');
```

Reply:
hm. i tested it (except pasted then in wrong place).  I'll try to find out what's going on, sorry

Reply:
hm... @kulikov-a I've just tested it again and now `toggle` works.... 😄 Maybe my browser cache was somehow not refreshed... 😞 
Sorry for the confusion...

Reply:
upd. take whole .volt form pr and tested on chrome, opera, edge, ff and ie: works well everywhere. can you please try to replace whole .volt? (maybe there is something left from previous commits? cannot understand what can trigger a `click` on page load)

Reply:
@8191 hooray! so we're ready to show it to Ad?

Reply:
@kulikov-a yeah, looks good from my side as well! 
Thanks!